app/Http/Controllers/CategoryController.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Category;
+use Categories;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Lang;
+
+class CategoryController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function __construct(Category $category)
+    {
+        $this->category = $category;
+    }
+
+    public function index()
+    {
+        $categories = Category::latest()->paginate(config('app.paginate'));
+
+        return view('Admin.category.index', compact('categories'));
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        $list_category = $this->category->all();
+
+        return view('Admin.category.create', compact('list_category'));
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        $category = Category::create($request->all());
+
+        return redirect()->route('categories.index')->with(
+            'message',
+            ['msg' => trans('create_category_success'), 'status' => 'success']
+        );
+    }
+
+    /**
+     * Show the form for editing the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function edit($id)
+    {
+        $category = $this->category->find($id);
+        $categories = Category::whereNull('parent_id')->get();
+
+        return view('Admin.category.edit', compact('categories', 'category'));
+    }
+
+    public function getCaregory($parent_id)
+    {
+        $list_category = $this->category->all();
+    }
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function update(Request $request)
+    {
+        $category = $this->category->update($request->all());
+        if (!$category) {
+            return redirect()->route('categories.index')->with(
+                'message',
+                ['msg' => trans('update_category_success'), 'status' => 'success']
+            );
+        }
+        return redirect()->route('categories.index')->with(
+            'message',
+            ['msg' => trans('update_category_error'), 'status' => 'danger']
+        );
+    }
+
+    /**
+     * Remove the specified resource from storage.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function destroy($id)
+    {
+        $category = $this->category->delete($id);
```

Comment:
Cần check $this->category có tồn tại category ID đó không đã rồi mới gọi hàm xóa
Chứ không null mà gọi ->delete là tèo
Có thể đỡ if else nhiều rườm rà thì đặt trong try catch em nhé

Reply:
dạ anh